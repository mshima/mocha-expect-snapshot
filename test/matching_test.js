const { expect } = require('expect');

const updateTestSnapshot = process.env.UPDATE_TEST_SNAPSHOT === 'true';

describe('matching test', function () {
  if (updateTestSnapshot) {
    before(function () {
      this.snapshotStateOptions = {
        updateSnapshot: 'all',
      };
    });
  }

  describe('toMachSnapshot()', function () {
    it('should match one snapshot', function () {
      expect({ foo: 'bar' }).toMatchSnapshot();
    });

    it('should match two snapshots', function () {
      expect({ foo: 'bar' }).toMatchSnapshot();
      expect({ foo2: 'bar' }).toMatchSnapshot();
    });

    it('should match error snapshots', function () {
      expect(() => {
        throw new Error('foo');
      }).toThrowErrorMatchingSnapshot();
    });

    it('should match inline error snapshots', function () {
      expect(() => {
        throw new Error('foo');
      }).toThrowErrorMatchingInlineSnapshot(`"foo"`);
    });
  });
});
