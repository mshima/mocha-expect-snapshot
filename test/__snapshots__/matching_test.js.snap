// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matching test toMachSnapshot() should match error snapshots 1`] = `"foo"`;

exports[`matching test toMachSnapshot() should match one snapshot 1`] = `
Object {
  "foo": "bar",
}
`;

exports[`matching test toMachSnapshot() should match two snapshots 1`] = `
Object {
  "foo": "bar",
}
`;

exports[`matching test toMachSnapshot() should match two snapshots 2`] = `
Object {
  "foo2": "bar",
}
`;
